src/FavIcon.vala

```diff
@@ -37,14 +37,24 @@ public class FeedReader.FavIcon : GLib.Object
 	private Gee.Promise<Gdk.Pixbuf?> m_pixbuf = null;
 	private ResourceMetadata m_metadata;
 
-	public signal void pixbuf_changed(Feed feed, Gdk.Pixbuf pixbuf);
+	public signal void surface_changed(Feed feed, Cairo.Surface surface);
 
 	private FavIcon(Feed feed)
 	{
 		m_feed = feed;
 	}
 
-	public async Gdk.Pixbuf? get_pixbuf()
+	private int GetScaleFactor()
+	{
+		return MainWindow.get_default().get_style_context().get_scale();
+	}
+
+	private Cairo.Surface CreateSurfaceFromPixbuf(Gdk.Pixbuf pixbuf)
```

Comment:
Note: I thought we wanted names like `create_surface_from_pixbuf`?

Reply:
Oh okay, note taken.